metricbeat/mb/builders.go

```diff
@@ -22,39 +22,32 @@ var (
 	ErrAllModulesDisabled = errors.New("all modules are disabled")
 )
 
-// NewModules builds new Modules and their associated MetricSets based on the
-// provided configuration data. config is a list module config data (the data
+// NewModule builds a new Module and its associated MetricSets based on the
+// provided configuration data. config contains config data (the data
 // will be unpacked into ModuleConfig structs). r is the Register where the
 // ModuleFactory's and MetricSetFactory's will be obtained from. This method
-// returns a mapping of Modules to MetricSets or an error.
-func NewModules(config []*common.Config, r *Register) (map[Module][]MetricSet, error) {
-	if config == nil || len(config) == 0 {
-		return nil, ErrEmptyConfig
+// returns a the Modules and its configured MetricSets or an error.
+func NewModule(config *common.Config, r *Register) (Module, []MetricSet, error) {
+	if !config.Enabled() {
+		return nil, nil, nil
```

Comment:
WDYT about returning an error in this case? Without an error callers always need to check if Module is non-nil.